### DISCUSSION TITLE
After visiting of arena, all other arenas get "visited" status.

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#9924] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

![image](https://user-images.githubusercontent.com/55348946/190911350-ee7eb62b-ab64-4e23-9e57-3b58fcc410fa.png)
![image](https://user-images.githubusercontent.com/55348946/190911362-351940ef-e11d-42f7-8645-13b31dc21745.png)
The bug appeared in the OG too.

### Save file

[Arena.zip](https://github.com/ihhub/fheroes2/files/9594150/Arena.zip)

**Comment by ihhub** [comment]
Hi @Branikolog , what's about interaction with other arenas? Can a hero visit them and get bonuses?

**Comment by Branikolog** [comment]
Hello, @ihhub !

>Can a hero visit them and get bonuses?

That's the main reason I made the issue. :) No bonuses at all. ^_^

**Comment by ihhub** [comment]
Then what's the problem? In the original game allows to visit any arena only once and we do the same then I see no bug in it.

**Comment by Branikolog** [comment]
@ihhub 
Umhmmm. You can visit **only one** Arena on a particular map. Even if there're 20 arenas on a map, you'll get only one primary skill raised +1. Do you think, the limit of visiting only one arena per map for each hero is ok? Then I'll close the issue. :)

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog and @ihhub,

There is this strange way of working in the original game for several types of places that is a bit different from the way other similar places work: Arena, Magic Well, Watering Hole, Artesian Spring, Stables, Oasis to compare with Mercenary Camp, Fort, Witch Doctor's Hut, Standing Stones, etc.

I plan to adjust this in my mod: https://github.com/ihhub/fheroes2/discussions/3136, https://github.com/ihhub/fheroes2/discussions/3134, https://github.com/ihhub/fheroes2/discussions/3137, https://github.com/ihhub/fheroes2/discussions/3135, https://github.com/ihhub/fheroes2/discussions/3139, https://github.com/ihhub/fheroes2/discussions/3138. :)

> @ihhub Umhmmm. You can visit **only one** Arena on a particular map. Even if there're 20 arenas on a map, you'll get only one primary skill raised +1. Do you think, the limit of visiting only one arena per map for each hero is ok? Then I'll close the issue. :)

Yes, this is also my point, that's why I would like these changes :)

**Comment by ihhub** [comment]
Magellan's Map object works exactly the same way. Are there any official maps with multiple Arenas? What does the manual state about this object?

**Comment by ihhub** [comment]
Also Sirens and Hut of Magi objects work the same way so I do not fully believe that this was a mistake.

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,

> Magellan's Map object works exactly the same way.

This is different.
The Magellan's Map object removes the fog of war for the entire sea portion of the adventure map.
Once this is done, it cannot be done again.

> Are there any official maps with multiple Arenas?

I'll check, but I think there are some.

> What does the manual state about this object?

The manual is broken, it shows the image of the Coliseum of the Barbarian castle. :)

![Broken manual arena 01](https://user-images.githubusercontent.com/43583994/190917708-faed738b-3ddd-4fa2-b99a-35d409dcb43a.png)

> Also Sirens and Hut of Magi objects work the same way so I do not fully believe that this was a mistake.

Hut of Magi is the same case as Magellan's Map, once this is done, it cannot be done again.
Sirens locations should also be independent and should be visitable several times: https://github.com/ihhub/fheroes2/discussions/3132.

**Comment by Branikolog** [comment]
@ihhub 
After all, it's the original behavior of this object.
If you think, it was intended to make such limitation, so it could be left as it is. I'm not against. 
We can later add another type of object  that would allow a hero to visit such type of object for multiple times. 
The logic of arena is not clear, so it would be nice to add the description, that hero could visit only one arena per game.

**Comment by LeHerosInconnu** [comment]
@Branikolog and @ihhub,

Basically, the Arena is just either a Mercenary Camp, a Fort, or Standing Stones, at the player's choice.
I don't see why it should be limited to one time per map, it's not very logical.
If this should be the case, the placement should have been limited to one copy on the aventure map in the scenario editor as it is for the ultimate artifact.

**Comment by zenseii** [comment]
> @Branikolog and @ihhub,
> 
>  If this should be the case, the placement should have been limited to one copy on the aventure map in the scenario editor as it is for the ultimate artifact.

You might want to have several arenas so that more players can get access to them early on in their own area etc.

I support @Branikolog's suggestion for this one.

**Comment by ihhub** [comment]
In this case I would insist to get rid of the corresponding experimental option and make everything standardized.

**Comment by LeHerosInconnu** [comment]
Hello @ihhub, @zenseii and @Branikolog,

> > Are there any official maps with multiple Arenas?
>
> I'll check, but I think there are some.

Unless I am mistaken, there are several Arenas in the eighth scenario of the PoL campaign and in the fourth scenario of the Wizard's Isle campaign.

> > @Branikolog and @ihhub,
> > If this should be the case, the placement should have been limited to one copy on the aventure map in the scenario editor as it is for the ultimate artifact.
> 
> You might want to have several arenas so that more players can get access to them early on in their own area etc.
> 
> I support @Branikolog's suggestion for this one.

The bonus is only one point in one of the primary skills.
It's like a Mercenary Camp, a Fort, a Witch Doctor's Hut or Standing Stones.
It's not a phenomenal bonus.
For example, Xanadu gives one point in each of the primary skills and every Xanadu on the map can be visited to get the bonus, so it's cumulative.

This is why @Branikolog opened the issue in the first place, because it seemed strange that each Arena could not be visited independently to accumulate the bonus.
And that's what I think too.

> In this case I would insist to get rid of the corresponding experimental option and make everything standardized.

Indeed, the Arena should offer the four primary skills by default and the option could be removed.
Since the player still has the possibility to ignore the choice of the primary skill Knowledge to have the same operation as the original game.
I always activate this option when I play.

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu 

True, the operation seemed strange, considering other similar primary skill rising objects. I think no one of true H2 fans won't argue with that.

> Since the player still has the possibility to ignore the choice of the primary skill Knowledge to have the same operation as the original game.

But missing knowledge for me is not such a problem. I consider arena as a regular object rising only 3 primary skills. I think this was made intetively to limit knowledge level of might/power based heroes.

**Comment by zenseii** [comment]
Hi, @LeHerosInconnu and @Branikolog 
Personally I think it is in line with the story of the arena to not teach you knowledge since it sounds weird that the gladiator master should agree to lecture you about books, which is what the Knowledge icon would suggest. Spell power is different and more "combat" type skill and I can imagine you training the mind.

I'm for the idea of having a separate object that gives all 4 or something of the sort and potentially be visited more times.

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog,

> Hi, @LeHerosInconnu
> 
> True, the operation seemed strange, considering other similar primary skill rising objects. I think no one of true H2 fans won't argue with that.

Especially if the Arena is protected by wandering monsters and the player realizes that the hero has already visited another Arena and his efforts were in vain. :)

> > Since the player still has the possibility to ignore the choice of the primary skill Knowledge to have the same operation as the original game.
> 
> But missing knowledge for me is not such a problem. I consider arena as a regular object rising only 3 primary skills. I think this was made intetively to limit knowledge level of might/power based heroes.

That's why I like the possibility to choose Knowledge when the hero is a bit weak in Knowledge. :)
So you think it should stay as an option?
Because for me, it's really an improvement to be able to choose Knowledge.

**Comment by LeHerosInconnu** [comment]
Hello @zenseii,

> Hi, @LeHerosInconnu and @Branikolog
Personally I think it is in line with the story of the arena to not teach you knowledge since it sounds weird that the gladiator master should agree to lecture you about books, which is what the Knowledge icon would suggest. Spell power is different and more "combat" type skill and I can imagine you training the mind.

Power is nothing without Knowledge. :rofl: 
 
> I'm for the idea of having a separate object that gives all 4 or something of the sort and potentially be visited more times.

Adding a location with a similar effect to another one just to preserve the functioning of the original game location is a bit overkill.
In this case it would be better to keep the option already present (Knowledge choice) and add another option to accumulate bonuses when the hero visits several Arena.

**Comment by zenseii** [comment]
Hi @LeHerosInconnu !

> Adding a location with a similar effect to another one just to preserve the functioning of the original game location is a bit overkill. In this case it would be better to keep the option already present (Knowledge choice) and add another option to accumulate bonuses when the hero visits several Arena.

The problem is that we don't have the "option" (😂) to add lots of options for single adventure map objects etc.. @ihhub wrote a very good clear wiki article explaining the idea: https://github.com/ihhub/fheroes2/wiki/Adding-a-new-option-into-the-game. 
So having such an option as you're mentioning doesn't fit into this criteria since it doesn't follow point 5 from that article in my opinion, at least you won't see such an option within the game's settings.

However, such options could very likely become possible in the future when the modding side of the project materializes, like being able to customize adventure map object functions and dwellings among other things. I think this would be the best compromise for everyone?

Still, seeing how contested the option of having the choice of all 4 skills in the arena is, maybe it's best to hold out on removing it as of now? That is, if the plan is at all to remove and not to make it the default behavior for the arena.

**Comment by ihhub** [comment]
Moving this into the Discussion section as this is a modification of the original game's behavior and some maps depend on it.
